File: integration-test/src/test/java/com/promise/integrationtest/server/dto/PostServerServerGroupResponse.java
package com.promise.integrationtest.server.dto;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.promise.integrationtest.base.ResourceResponse;

public class PostServerServerGroupResponse extends ResourceResponse
{
    @JsonProperty(value = "ServerID", required = true)
    private String serverId;
    @JsonProperty(value = "ServerGroupID", required = true)
    private String serverGroupId;
    @JsonProperty(value = "ServerURI", required = true)
    private String serverUri;
    @JsonProperty(value = "ServerGroupURI", required = true)
    private String serverGroupUri;

    public String getServerId()
    {
        return serverId;
    }

    public void setServerId(String serverId)
    {
        this.serverId = serverId;
    }

    public String getServerGroupId()
    {
        return serverGroupId;
    }

    public void setServerGroupId(String serverGroupId)
    {
        this.serverGroupId = serverGroupId;
    }

    public String getServerUri()
    {
        return serverUri;
    }

    public void setServerUri(String serverUri)
    {
        this.serverUri = serverUri;
    }

    public String getServerGroupUri()
    {
        return serverGroupUri;
    }

    public void setServerGroupUri(String serverGroupUri)
    {
        this.serverGroupUri = serverGroupUri;
    }

}

File: server/object/dto/PostServerServerGroupResponse.go
package dto

import (
	"promise/common/object/constValue"
	"promise/server/object/model"
)

// PostServerServerGroupResponse is the DTO.
type PostServerServerGroupResponse struct {
	ResourceResponse
	ID             string `json:"ID"`
	URI            string `json:"URI"`
	ServerID       string `json:"ServerID"`
	ServerURI      string `json:"ServerURI"`
	ServerGroupID  string `json:"ServerGroupID"`
	ServerGroupURI string `json:"ServerGroupURI"`
}

// Load the data from model.
func (dto *PostServerServerGroupResponse) Load(m *model.ServerServerGroup) {
	dto.ID = m.ID
	dto.URI = constValue.ToServerServerGroupURI(m.ID)
	dto.ServerID = m.ServerID
	dto.ServerURI = constValue.ToServerURI(m.ServerID)
	dto.ServerGroupID = m.ServerGroupID
	dto.ServerGroupURI = constValue.ToServerGroupURI(m.ServerGroupID)
}
